src/transform/datastructures.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (C) 2021 NV Access Limited
+# This file may be used under the terms of the GNU General Public License, version 2 or later.
+# For more details see: https://www.gnu.org/licenses/gpl-2.0.html
+
+from dataclasses import dataclass
+from typing import Dict, Literal, NamedTuple, Optional, OrderedDict, Union
+
+AddonChannels = Literal["beta", "stable"]
+
+
+class MajorMinorPatch(NamedTuple):
+	""" The MajorMinorPatch datastructure is used by NVDAVersions and AddonVersions.
+	Ensure that if a patch number is specified, it is preserved in strings, otherwised ignored.
+	For example NVDAVersion(2019, 1) should not end up creating paths like /2019.1.0/.
+	However, AddonVersion.fromStr("1.3.0") should preserve the patch number.
+	"""
+	major: int
+	minor: int
+	patch: Union[int, bool] = False  # is compared as 0, but works as a flag for hiding the patch by default
+
+	@classmethod
+	def fromStr(cls, version: str):
+		versionTuple = tuple(int(v) for v in version.split("."))
+		return MajorMinorPatch(*versionTuple)
```

Comment:
I think it would be good to be consistent with the version number scheme. Either `src/validate/addon_data.schema.json` should be updated to use the "canonical" version number used in the submissions schema, or the submissions schema should be updated to use a string.

Using 3 integers allows the json library to do the work of validation and separating the parts (at the expense of greater verbosity in the file). I suspect that these files will be created by a form on a GH issue eventually, so I'm not so worried about the verbosity.

Reply:
This was written before that was introduced and we had to rely on the file name. I've just moved this code to the NVDAVersion datastructure.

Reply:
Still, I think either this and NVDAVersions.json should be updated to use 3 integers, or the other repositories should be updated to use strings and regex.